### DISCUSSION TITLE
Telescope git_status git_staging_toggle

### DISCUSSION THREAD

**Discussion by krishna-bala** [LazyVim/LazyVim#646] [Q&A] [answered]
It seems as though the toggle action for `:Telescope git_status` (default mapping: `<leader>gs`) is not fully working. The toggle action is mapped to `<Tab>` when in the `git_status` picker by default and `<C-?>` confirms that mapping. 

I am able to toggle an unstaged file (move from unstaged to staged), but I am unable to toggle a staged file to unstaged. I have tested this with the default LazyVim installation, but have not tested if this issue is a telescope issue (i.e. I have not tried using telescope on its own outside of the LazyVim setup).

Does anyone else have this issue? I suspect this a telescope issue and not a LazyVim issue, but if anyone knows otherwise that would be helpful.

**Comment by dpetka2001** [comment]
I've never used `Telescope git status` before as I prefer Lazygit, but I just tested it out and I am able to toggle both directions (unstaged->staged and staged->unstaged with `<Tab>`). So I can't confirm the problem you're facing. I confirmed the change of status of the file from both Lazygit and `git status` on the command line.

**Comment by folke** [comment]
Works fine for me as well

**Comment by krishna-bala** [comment, marked as answer]
@dpetka2001 @folke thank you for confirming, very much appreciated! After inspecting [telescope/actions/init.lua:814](https://github.com/nvim-telescope/telescope.nvim/blob/master/lua/telescope/actions/init.lua#L814), I realized I am using an old version of git (`git version 2.17.1`) that does not have the `git restore` command. I am on Ubuntu 18.04.

Taking the following action solved the problem for me:
```
$ git --version
git version 2.17.1
$ sudo add-apt-repository ppa:git-core/ppa
$ sudo apt-get update
$ sudo apt-get install git
$ git --version
git version 2.40.0
```
